nibabel/gifti/gifti.py

```diff
@@ -501,7 +530,7 @@ def getArraysFromIntent(self, intent):
 
     def print_summary(self):
         print('----start----')
-        print('Source filename: ', self.filename)
+        print('Source filename: ', self.get_filename())
```

Comment:
`GiftiImage` was storing `filename` on it's own. It should use the mechanisms in place for `FileBasedImage`.